test/forge/HalmosTest.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import "../../lib/forge-std/src/Test.sol";
+import {SymTest} from "../../lib/halmos-cheatcodes/src/SymTest.sol";
+
+import {IMorpho} from "../../src/interfaces/IMorpho.sol";
+import {IrmMock} from "../../src/mocks/IrmMock.sol";
+import {ERC20Mock} from "../../src/mocks/ERC20Mock.sol";
+import {OracleMock} from "../../src/mocks/OracleMock.sol";
+
+import "../../src/Morpho.sol";
+import "../../src/libraries/ConstantsLib.sol";
+import {MorphoLib} from "../../src/libraries/periphery/MorphoLib.sol";
+
+/// @custom:halmos --symbolic-storage --solver-timeout-assertion 0
+contract HalmosTest is SymTest, Test {
+    using MorphoLib for IMorpho;
+    using MarketParamsLib for MarketParams;
+
+    uint256 internal constant BLOCK_TIME = 1;
+
+    address internal OWNER;
+    address internal FEE_RECIPIENT;
+
+    IMorpho internal morpho;
+    ERC20Mock internal loanToken;
+    ERC20Mock internal collateralToken;
+    OracleMock internal oracle;
+    IrmMock internal irm;
+
+    MarketParams internal marketParams;
+    Id internal id;
+
+    function setUp() public virtual {
+        OWNER = address(0x10);
+        FEE_RECIPIENT = address(0x11);
+
+        morpho = IMorpho(address(new Morpho(OWNER)));
+
+        loanToken = new ERC20Mock();
+        collateralToken = new ERC20Mock();
+        oracle = new OracleMock();
+        oracle.setPrice(ORACLE_PRICE_SCALE);
+        irm = new IrmMock();
+
+        vm.startPrank(OWNER);
+        morpho.enableIrm(address(0));
+        morpho.enableIrm(address(irm));
+        morpho.enableLltv(0);
+        morpho.setFeeRecipient(FEE_RECIPIENT);
+        vm.stopPrank();
+
+        uint256 lltv = svm.createUint256("lltv");
+        marketParams = MarketParams(address(loanToken), address(collateralToken), address(oracle), address(irm), lltv);
+        id = marketParams.id();
+
+        vm.assume(block.timestamp != 0);
+        vm.startPrank(OWNER);
+        morpho.enableLltv(lltv);
+        morpho.createMarket(marketParams);
+        vm.stopPrank();
+
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 1 * BLOCK_TIME);
+    }
+
+    // Call Morpho, assuming interacting with only the defined market for performance reasons.
+    function _callMorpho(bytes4 selector, address caller) internal {
+        vm.assume(selector != morpho.extSloads.selector);
+        vm.assume(selector != morpho.createMarket.selector);
```

Comment:
creating a new market doesn't break rules, just break the prover right (timeout)?

Reply:
It does break the rules, see [this comment](https://github.com/morpho-org/morpho-blue/pull/675#discussion_r1574685906)

Reply:
`createMarket` will fail if we pass symbolic values to it. This is because `createMarket` calls the IRM, and calling a symbolic address is not permitted. Because of this, I found it cleaner to avoid calling `createMarket` altogether

Reply:
but actually a lot of calls that we do fail and the prover is ok with that no?

Reply:
i understand now, calls to arbitrary addresses are not supported

Reply:
Yes but there are 2 kinds of failures: either a revert or a unsupported situation by Halmos. In the latter, it just reports as the test failing. For reverts, Halmos assumes that the path is not taken. This is similar to what Certora is doing (except if you add the `@withrevert` suffix to the function call). This means that you can verify functions that revert for some paths, and verify the property for all of the paths that do not revert.
Here calling a symbolic address is unsupported by Halmos, so the test would always show a fail